### PR TITLE
Fix deprecation warnings for imports from collections

### PR DESCRIPTION
**Pull request by bdrung**
DeprecationWarning: Using or importing the ABCs from `collections` instead of from `collections.abc` is deprecated since Python 3.3, and in 3.9 it will stop working.

Therefore try to import the abstract base classes from `collections.abc` before falling back to `collections`.

### CODE REVIEW

**Comment by mchugh19** [comment]
This looks to be a superset of #55900

**Comment by bdrung** [comment]
Right. I wasn't aware that there is already a similar pull request. I just grepped for `DeprecationWarning` in the unit test output and fixed all occurrences.

**Review by twangboy — DISMISSED** [review]

**Review by DmitryKuzmenko — DISMISSED** [review]

**Comment by DmitryKuzmenko** [comment]
re-run full ubuntu1604-py3-pycryptodomex
re-run full centos7-py2-tornado
re-run full centos7-py2
re-run full amazon2-py2

**Comment by DmitryKuzmenko** [comment]
@bdrung could you please resolve merge conflicts?

**Comment by Ch3LL** [comment]
ping @DmitryKuzmenko and @twangboy looks like this is waiting on your approval

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]
I'm sorry for later response and for asking you for another one change. This probably wasn't officially introduced but `salt/ext` contains foreign code that we're keeping as close to the original as possible to don't have much troubles updating it. So could you please revert or rebase dropping your last commit e1f57d6207ba4e6d30f9d7c69c52398dc98065ca that is blacking the `salt/ext/tornado/httputil.py`.
Just leave your `collections` rfelated` changes there.
Thank you for understanding.

**Review by twangboy — DISMISSED** [review]

**Comment by bdrung** [comment]
I removed e1f57d6 and rebased the branch to fix the merge conflicts. Expect some linter checks to fail after this removal.

**Pull request merged**
